### ISSUE TITLE
Allow coordinators to set default project data

### ISSUE THREAD

**Issue by sagesharp** [sagesharp/outreachy-django-wagtail#295]
Some projects in the same community have common data across all projects (like the Linux kernel). Other communities have projects that differ widely (for example Mozilla's projects).

Allow coordinators to set defaults for project information. The defaults will be displayed when the mentor fills out the form. Coordinators should be able to set defaults for:

- [ ] How to make their first contribution. For example, the Linux kernel community has a tutorial that everyone should go through before starting on project-specific tasks. Mentors can always add additional information to that text.
- [ ] Minimum system requirements
- [ ] Project repository
- [ ] Issue tracker
- [ ] Newcomer issue tag

The above fields are pretty easy to add defaults for, since they're all in one Django model (class Project in home/models.py). There are other things we'd like to set defaults for which aren't so simple to add for technical reasons. (They have a many-to-one relationship to the Project object, which makes it harder to fill in defaults for them in the forms.)

Here's additional items that are stretch goals for coordinators to set defaults for:

- [ ] Project skills
- [ ] Project communication channels

**Comment by jameysharp** [comment]
I thought this was the documentation you'd want:

- https://docs.djangoproject.com/en/1.11/topics/forms/modelforms/#providing-initial-values
- https://docs.djangoproject.com/en/1.11/ref/forms/api/#dynamic-initial-values

But the modelforms documentation says that `initial` values override not just the field default, but also the value in an initial model instance, if you have one. I think we're using an `UpdateView` for `Project` so that the same view can work for both creation (where `initial` does what you want) and modification of existing projects (where `initial` would overwrite the previously-specified values of fields, which is bad).

So I guess the right thing to do instead is to change the `get_object` method in the view. If trying to get an existing `Project` raises `Project.DoesNotExist`, then just pass the custom default values to the `Project` constructor. (This has the added advantage of not needing to use any more Django features than we already do. :smile_cat:)

**Comment by sagesharp** [comment]
For default communication channels, we need to create them after the project has been saved (so that it has a pk we can link the communication channel to). We can add the default channels objects in the `get_success_url()` method of ProjectAction (which is the function where projects are created or updated). We already create a new MentorApproval object if one doesn't exist, so we can create additional ProjectSkill and CommunicationChannel objects there too. See:

https://github.com/sagesharp/outreachy-django-wagtail/blob/master/home/views.py#L1365

**Comment by sagesharp** [comment]
Talking with @jameysharp about the technical details, I think we have a plan:

- The default project settings will be stored in a Project associated with the round the coordinator signed up in.
- The project will have an empty string for the project title, leading to an empty slug
- That means all normal project URLs (such as recording a contribution for that project) won't work. Editing of the default project settings will happen in a new view.
- When a coordinator signs up to participate in a new round, we copy (or move) the default project to be associated with the new round
- After the coordinator signs up to participate, the next page will ask them to set the default information for all projects
- We'll have to look at all the views that display projects (or functions that return projects) and make sure to exclude the project with an empty slug

This has the added benefit that we don't need to add any new models. Only views need to be changed, and a couple new templates need to be added.
